Destroy pools even when they are not loaded

tiagolobocastro opened the pull request:
chore: disable tracing for registration

This is too verbose as they are sent every x seconds.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

fix: import pool when destroying

When a node is restarted, we may issue a pool destroy before the reconciler imports it.
Note: the reconciler will never attempt to import it after this as the pool will now be in
the Destroying phase.
To address this, if pool destroy fails with status NotFound, we then attempt to import it
and then retry the destruction.

Added a test case for this.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

refactor: tidy-up docs comments for pool definitions

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

fix(pool/operator): make delete idempotent

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: add info traces for completed operations

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

 fix: attempt to decrypt not found errors for pool creation

Really the dataplane should return proper status codes rather than internal, this is
an interim step..

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: disable tracing for registration

This is too verbose as they are sent every x seconds.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix: import pool when destroying

When a node is restarted, we may issue a pool destroy before the reconciler imports it.
Note: the reconciler will never attempt to import it after this as the pool will now be in
the Destroying phase.
To address this, if pool destroy fails with status NotFound, we then attempt to import it
and then retry the destruction.

Added a test case for this.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: add info traces for completed operations

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

Abhinandan-Purkait commented on control-plane/agents/src/bin/core/pool/pool_operations.rs:
```diff
@@ -62,7 +63,15 @@ impl ResourceLifecycle for OperationGuardArc<PoolSpec> {
+            Ok(_) => Ok(()),
+            Err(SvcError::PoolNotFound { .. }) => {
+                node.import_pool(&self.as_ref().into())
+                    .and_then(|_| async { node.destroy_pool(request).await })
```
👍🏻

Abhinandan-Purkait commented on control-plane/agents/src/lib.rs:
```diff
@@ -62,7 +62,12 @@ impl Service {
+                // This is a bit of hack, but tonic doesn't seem to provide access to this uri
+                // path in any way.
+                // todo: add ignored routes via shared state
+                "/mayastor.v1.Registration/Register",
```
Question: Why do we want to ignore registration traces?

tiagolobocastro replied:
They're too verbose

tiagolobocastro replied:
An important detail, these are not the traces but rather the top-level spans.

Abhinandan-Purkait replied:
Right. Thanks

Abhinandan-Purkait commented on control-plane/stor-port/src/types/v0/transport/pool.rs:
```diff
@@ -272,6 +272,32 @@ impl CreatePool {
     }
 }
 
+/// Create Pool Request
```
`.` missing at the end? Also the field doc comments should be converted to the format?

tiagolobocastro replied:
I'd need to change entire file... I'll do it

Abhinandan-Purkait replied:
Thanks

tiagolobocastro committed:
fix(pool/operator): make delete idempotent

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
refactor: tidy-up docs comments for pool definitions

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
fix: attempt to decrypt not found errors for pool creation

Really the dataplane should return proper status codes rather than internal, this is
an interim step..

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge
